data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/s3/S3SourceConfig.java

```diff
@@ -43,6 +44,10 @@ public class S3SourceConfig {
     @Valid
     private SqsOptions sqsOptions;
 
+    @JsonProperty("workers")
+    @Valid
+    private int numWorkers = DEFAULT_NUMBER_OF_WORKERS;
```

Comment:
Please add an `@Min(1)` on this. 

We should probably also have some sort of upper bound to prevent excessive values. Perhaps: `@Max(1000)`.